### PR TITLE
Riem solver3 is an object

### PR DESCRIPTION
**Pull request by rheacangeo**
## Purpose

Use our OOP conventions to make the riemann solver3 a class. 

## Code changes:

- stencils/riem_solver3, the test code that calls it 
- removed temporaries in riem_solver3 that were not needed
- dyn_core uses a cached stencil class initialization to call it
- the sim1_solver class sets the fixed origin in class initialization rather than runtime

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate

### CODE REVIEW

**Comment by rheacangeo** [comment]
@eddie-c-davis I know you are working with the parallel riem_solverc so please let me know if there are any conventions you are adopting I should shift to here. 
@FlorianDeconinck I made a small change to the sim1_solver you just merged so thought you might want to give feedback on it

File: fv3core/stencils/sim1_solver.py [hunk 1]
```diff
@@ -123,9 +123,11 @@ def __init__(self, namelist, grid, istart, iend, jstart, jend):
...
         nic = iend - istart + 1
         njc = jend - jstart + 1
-        self._origin = (istart, jstart, 0)
-        self._domain = (nic, njc, grid.npz + 1)
-        self._compute_sim1_solve = StencilWrapper(func=sim1_solver, externals={})
+        self._compute_sim1_solve = StencilWrapper(
+            func=sim1_solver,
+            origin=(istart, jstart, 0),
+            domain=(nic, njc, grid.npz + 1),
+        )
```
Comment (FlorianDeconinck) [inline fv3core/stencils/sim1_solver.py:130]: Good catch, I figured this one on the other object I refactored :D

File: fv3core/stencils/dyn_core.py [hunk 2]
```diff
@@ -527,15 +527,16 @@ def __call__(self, state):
                     state.wsd,
                     dt,
                 )
-
-                riem_solver3.compute(
+                riem_solver3 = utils.cached_stencil_class(RiemannSolver3)(
```
Comment (FlorianDeconinck) [inline fv3core/stencils/dyn_core.py:530]: You shouldn't need the caching. you should be able to move that to __init__

**Review by FlorianDeconinck — APPROVED** [review]
Could you expand my rather slim `sim_solver1` docstring with what it's actually doing? I couldn't figure it out

**Review by eddie-c-davis — APPROVED** [review]
Excellent work! Looks like Florian caught anything I noticed.

File: fv3core/stencils/dyn_core.py (hunk 2 above)
Comment (rheacangeo) [reply fv3core/stencils/dyn_core.py:530]: good catch, I forgot dyn_core was an abject, even though I'm the one that made it so!

**Comment by rheacangeo** [comment]
launch jenkins

**Comment by rheacangeo** [comment]
eeks after a super weird error gt4py/_external_src/gridtools/include/gridtools/stencil_composition/sid/composite.hpp:31:28: error: redefinition of 'struct gridtools::sid::composite::impl_::lazy::make_map_helper<State, Kind>'
        struct make_map_helper {
when it was trying to recompile precompute ... trying again worked. weird.

**Pull request merged**